app/views/instructeurs/avis/all.html.haml

```diff
@@ -0,0 +1,39 @@
+- content_for(:title, "Avis")
+
+.container
+  %h1.page-title Avis
+
+  %ul.procedure-list
+    - @procedures.each do |p|
+      %li.procedure-item.flex.align-start
+        = link_to(instructeur_avis_index_path(p)) do
+          .flex
+
+            .procedure-logo{ style: "background-image: url(#{p.logo_url})" }
+
+            .procedure-details
+              %p.procedure-title
+                = procedure_libelle p
+              %ul.procedure-stats.flex
+                - procedure_avis = current_instructeur.avis.includes(:dossier).where(dossiers: { groupe_instructeur: GroupeInstructeur.where(procedure: p.id) })
+                %li
+                  %object
+                    = link_to(instructeur_avis_index_path(p, statut: Instructeurs::AvisController::A_DONNER_STATUS)) do
+                      .stats-number
+                        = procedure_avis.without_answer.count
```

Comment:
J'imagine qu'il y a moyen côté controlleur d'éviter de faire ces n+1 queries, à l'image de ce qui est fait dans `ProceduresController#index` mais je n'ai pas réussi à dire à activerecord de faire le group_by qui va bien :(

https://github.com/betagouv/demarches-simplifiees.fr/blob/dev/app/controllers/instructeurs/procedures_controller.rb#L18

Reply:
Mmm, effectivement je ne vois pas trop…

Ah si, j'ai peut-être une idée :

Peut-être que ce qui nous induit en erreur, c'est que la méthode du contrôller s'appelle `all` (vu que `index` est déjà utilisé).

Mais en fait, cette nouvelle page qui affiche tous les avis, ça devrait être elle qui est nommée `index`. Et le contrôleur devrait faire une requête pour récupérer tous les **Avis** (plutôt que des Procedures), avec un `includes(:procedure)` – et ensuite en Ruby (ou en SQL) grouper ces avis par procédure.

Comme ça ensuite la vue peut simplement énumérer les avis-groupés-par-procédure, et afficher la liste sans faire de N+1.

Du coup ça voudrait dire :

- Renommer `AvisController#index` -> `AvisController#procedure`
- Renommer `AvisController#all` -> `AvisController#index`
- Faire en sorte que `AvisController#index` fasse qqc du genre `@avis = current_instructeur.avis.includes(:procedure).group_by(:procedure_id)` (ou la même chose mais en groupant en Ruby)

Tu en penses quoi ?

Reply:
D'accord avec ton renommage qui me parait plus railish :)

Pour l'idée d'énumérer les avis groupés par procédure à partir de `current_instructeur.avis`, j'avais essayé mais sans succès.

Le truc, c'est que l'avis et la procédure sont très éloignés dans l'arbre.
ActiveRecord arrive à interpréter ceci :  
` avis = current_instructeur.avis.includes(dossier: [groupe_instructeur: :procedure])`

Mais je n'arrive pas ensuite à dire à activerecord de grouper par procedure_id.
`avis.group('groupe_instructeurs.procedure_id')` me pète cette exception : 

`PG::UndefinedTable: ERROR:  missing FROM-clause entry for table "groupe_instructeurs")`

Reply:
Ah oui… Et moi je ne suis pas assez doué en Active Record pour essayer de faire fonctionner ça.

Et sinon si on fait le groupage en Ruby ? Vu qu'on a déjà toutes les requêtes, ça devrait être raisonnable niveaux perfs, non ?

```ruby
avis = current_instructeur.avis.includes(dossier: [groupe_instructeur: :procedure])
avis_by_procedure = avis.to_a.group_by { |a| a.procedure }
```

Reply:
Yep, c'était ma solution de repli :) J'ai tenté de pousser le truc au max, au cas où.

Du coup, j'ai fait le groupage en ruby, par ici : https://github.com/betagouv/demarches-simplifiees.fr/pull/5306/commits/112351dea5c5ea59bc41d45ca015008cdf529b72

Reply:
Et pour le renommage, c'est par ici que ça se passe : 

https://github.com/betagouv/demarches-simplifiees.fr/pull/5306/commits/bf6e10053fc12ce36d7824e68e1fa3f7f8ea9933

https://github.com/betagouv/demarches-simplifiees.fr/pull/5306/commits/1b823168b53f00ca3c99d7e420b62c33cd9e974c